File: coding-test/nodejs/src/kakao-blind-recruitment-2018/n-number-game.js
// @ts-check

/**
 * @param {number} n
 * @param {number} t
 * @param {number} m
 * @param {number} p
 * @returns {string}
 */
function solution(n, t, m, p) {
  let answer = ''
  let str = ''
  let number = 0
  for (let i = p - 1; i < m * t; i += m) {
    while (str.length <= i) {
      str += number.toString(n).toUpperCase()
      number += 1
    }

    answer += str.charAt(i)
  }

  return answer
}

const n = 16,
  t = 16,
  m = 2,
  p = 1

console.log(solution(n, t, m, p))

File: practice-coding-test/java-kotlin/src/kr/co/programmers/java/kakao_blind_recruitment_2018/NNumberGame.java
package kr.co.programmers.java.kakao_blind_recruitment_2018;

public class NNumberGame {

    public static void main(String[] args) {
        int n = 2;
        int t = 4;
        int m = 2;
        int p = 1;
        System.out.println(new Solution().solution(n, t, m, p));
    }

    static class Solution {
        public String solution(int n, int t, int m, int p) {
            String answer = "";

            int number = 0;
            StringBuilder toNString = new StringBuilder();

            while(toNString.length() < t * m){
                toNString.append(Integer.toString(number++, n).toUpperCase());
            }

            for(int i = 0; i < t; i++){
                answer += toNString.charAt(p - 1 + m * i);
            }

            return answer;
        }
    }
}
